whylog_vim/input_reader/__init__.py

```diff
@@ -1,17 +1,48 @@
 import re
-from whylog_vim.input_reader.consts import Input, RegexPatterns
 
+from whylog.constraints import IdenticalConstraint
+from whylog.constraints.const import ConstraintType
 
-def filter_comments(content):
-    return [line for line in content if not RegexPatterns.COMMENTS.match(line)]
+from whylog_vim.input_reader.consts import RegexPatterns, ConstraintInput
+from whylog_vim.utils import get_id_from_name
 
 
-def get_button_name(line, offset):
-    offset -= 1
-    for match in re.finditer(RegexPatterns.BUTTON, line):
-        if offset >= match.start(0) and offset < match.end(0):
-            return match.group(0)[1:-1]
+class InputReader():
+    @classmethod
+    def filter_comments(cls, content):
+        return [line for line in content if not RegexPatterns.COMMENTS.match(line)]
 
+    @classmethod
+    def get_button_name(cls, line, offset):
+        offset -= 1
+        for match in re.finditer(RegexPatterns.BUTTON, line):
+            if offset >= match.start(0) and offset < match.end(0):
+                return match.group(0)[1:-1]
 
-def prepare_regex(pattern):
-    return ('^' + pattern + '$') % Input.EMPTY_GROUP
+    @classmethod
+    def parse_primary_key_groups(cls, content):
+        return map(int, content[0].split(', '))
+
+
+class ConstraintReader():
```

Comment:
inherit object. Do "git grep class" before PR assigning and try to catch those mistakes on your own, because it often repeats.